static/src/javascripts/projects/commercial/modules/dfp/dfp-api.spec.js

```diff
@@ -0,0 +1,422 @@
+            prepareGoogletag.init(noop, resolve);
+        })
+            .then(() => {
+                expect(Object.keys(getAdverts(true)).length).toBe(4);
+            })
+            .then(done)
+            .catch(done.fail);
+    });
+
+    it('should not get hidden ad slots', done => {
+        $('.js-ad-slot').first().css('display', 'none');
+        new Promise(resolve => {
+            prepareGoogletag.init(noop, resolve);
+        })
+            .then(() => {
+                const slots = getAdverts(true);
+                expect(Object.keys(slots).length).toBe(3);
+                Object.keys(slots).forEach(slotId => {
+                    expect(slotId).toBeTruthy();
+                    expect(slotId).not.toBe('dfp-ad-html-slot');
+                });
+            })
+            .then(done)
+            .catch(done.fail);
+    });
+
+    it('should set listeners', done => {
+        new Promise(resolve => {
+            prepareGoogletag.init(noop, resolve);
+        })
+            .then(() => {
+                expect(
+                    window.googletag.pubads().addEventListener
+                ).toHaveBeenCalledWith('slotRenderEnded', expect.anything());
+            })
+            .then(done)
+            .catch(done.fail);
+    });
+
+    it('should define slots', done => {
+        new Promise(resolve => {
+            prepareGoogletag.init(noop, resolve);
+        })
+            .then(() => {
+                [
+                    [
+                        'dfp-ad-html-slot',
+                        [[300, 50]],
+                        [[[0, 0], [[300, 50]]]],
+                        'html-slot',
+                    ],
+                    [
+                        'dfp-ad-script-slot',
+                        [[300, 50], [320, 50]],
+                        [[[0, 0], [[300, 50], [320, 50]]]],
+                        'script-slot',
+                    ],
+                    [
+                        'dfp-ad-already-labelled',
+                        [[728, 90], [300, 50], [320, 50]],
+                        [
+                            [[740, 0], [[728, 90]]],
+                            [[0, 0], [[300, 50], [320, 50]]],
+                        ],
+                        'already-labelled',
+                    ],
+                    [
+                        'dfp-ad-dont-label',
+                        [
+                            [728, 90],
+                            [900, 250],
+                            [970, 250],
+                            [300, 50],
+                            [320, 50],
+                        ],
+                        [
+                            [[980, 0], [[728, 90], [900, 250], [970, 250]]],
+                            [[740, 0], [[728, 90]]],
+                            [[0, 0], [[300, 50], [320, 50]]],
+                        ],
+                        'dont-label',
+                    ],
+                ].forEach(data => {
```

Comment:
Any chance to make these numbers less magic? We've got [chance](http://chancejs.com/) for creating random numbers, which could help in this case.